src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -217,6 +217,13 @@ private JsonConverter GetConverterFromAttribute(JsonConverterAttribute converter
             Debug.Assert(converter != null);
             if (!converter.CanConvert(typeToConvert))
             {
+                Type? underlyingType = Nullable.GetUnderlyingType(typeToConvert);
+                if (underlyingType != null)
+                {
+                    // Allow nullable handling to forward to the underlying type's converter.
+                    return JsonValueConverterNullableFactory.CreateValueConverter(underlyingType, converter);
```

Comment:
We should check if `converter.CanConvert(underlyingType)` returns true here before CreateValueConverter. If not, throw. (and add a test for that). This just means the converter is essentially applied to the wrong property.

Reply:
Good call, done.